patches/makefile.staticlink.patch

```diff
@@ -1,56 +1,4 @@
-632c632
```

Comment:
This diff of a patch is hard to read. However, I don't think we need this patch at all any more. It's original purpose was to remove `-lz` since we are statically linking. But with htslib 1.21, the `-lz`'s have all been removed already.

https://github.com/samtools/htslib/commit/c12496f3b16f441f7641fac62cd24db932cb81e5
https://github.com/samtools/htslib/pull/1828
https://github.com/samtools/htslib/compare/1.20...1.21#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52

Thus I think we should delete `patches/makefile.staticlink.patch` (and any references to it)

Reply:
Deleted.